client/state/action-types.js

```diff
@@ -78,6 +78,8 @@ export const COMMENTS_REPLY_WRITE = 'COMMENTS_REPLY_WRITE';
 export const COMMENTS_REQUEST = 'COMMENTS_REQUEST';
 export const COMMENTS_REQUEST_FAILURE = 'COMMENTS_REQUEST_FAILURE';
 export const COMMENTS_REQUEST_SUCCESS = 'COMMENTS_REQUEST_SUCCESS';
+export const COMMENTS_TREE_RECEIVE = 'COMMENTS_TREE_RECEIVE';
+export const COMMENTS_TREE_REQUEST = 'COMMENTS_TREE_REQUEST';
```

Comment:
Could it be instead that we use the same action type for all cases, adding more query params?
[Here](https://github.com/Automattic/wp-calypso/blob/631933e0f2ee4d1af73575242177381cdec11495/client/state/data-layer/wpcom/sites/comments/index.js#L124) for example, we could add a `query.listType` that could be `site`, `post` (and then require a `postId` as well), or `all`.